From 2e69532e79252bc2a0dcc51019350503a8414eed Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Thu, 5 Dec 2019 18:15:10 +0000
Subject: [PATCH 0902/1000] MANTIS 0055465 CUSS 1.5.1 Biometrics Interface

new version of CUSS-1.5.BIOMETRICS.GP.XSD
new version of CUSS.BIOMETRIC.XSD; referencing OneIdTypes.xsd


git-svn-id: svn://localhost/SelfServiceCommon/trunk@1506 90b65887-3827-0410-9a23-83215b262276
---
 .../idl/Cuss/CUSS-1.5.BIOMETRICS.GP.XSD       |   7 +-
 .../Massai/idl/Cuss/CUSS.BIOMETRIC.XSD        | 237 ++--------------
 .../Massai/idl/Cuss/OneIdTypes.xsd            | 268 ++++++++++++++++++
 3 files changed, 292 insertions(+), 220 deletions(-)
 create mode 100644 SelfServiceCommon/Massai/idl/Cuss/OneIdTypes.xsd

diff --git a/SelfServiceCommon/Massai/idl/Cuss/CUSS-1.5.BIOMETRICS.GP.XSD b/SelfServiceCommon/Massai/idl/Cuss/CUSS-1.5.BIOMETRICS.GP.XSD
index 43ca629c..e82afee1 100644
--- a/SelfServiceCommon/Massai/idl/Cuss/CUSS-1.5.BIOMETRICS.GP.XSD
+++ b/SelfServiceCommon/Massai/idl/Cuss/CUSS-1.5.BIOMETRICS.GP.XSD
@@ -143,6 +143,11 @@
           <xs:documentation>Optional status message returned by biometrics provider during operation execution (e.g. 'Success to register facial image (703)' or 'Face of camera image and passport image is different (-709)')</xs:documentation>
         </xs:annotation>
       </xs:enumeration>
+      <xs:enumeration value="materna.cuss.response.xb01CommandResult">
+        <xs:annotation>
+          <xs:documentation>High-level XB01 command result of the biometrics operation. This code is mapped from the biometrics backend error code to an XB01 command result (e.g. 'Face of camera image and passport image is different (-709)' -> 'matchFailed')</xs:documentation>
+        </xs:annotation>
+      </xs:enumeration>
     </xs:restriction>
   </xs:simpleType>
-</xs:schema>
+</xs:schema>
\ No newline at end of file
diff --git a/SelfServiceCommon/Massai/idl/Cuss/CUSS.BIOMETRIC.XSD b/SelfServiceCommon/Massai/idl/Cuss/CUSS.BIOMETRIC.XSD
index 72679ecb..c663ed22 100644
--- a/SelfServiceCommon/Massai/idl/Cuss/CUSS.BIOMETRIC.XSD
+++ b/SelfServiceCommon/Massai/idl/Cuss/CUSS.BIOMETRIC.XSD
@@ -1,219 +1,7 @@
-<xsd:schema xmlns:tns="urn:CUSS-1.5/types/biometrics" xmlns:xsd="http://www.w3.org/2001/XMLSchema" targetNamespace="urn:CUSS-1.5/types/biometrics" elementFormDefault="qualified" attributeFormDefault="unqualified">
-    <!-- Global types -->
-    <xsd:simpleType name="genderType">
-        <xsd:annotation>
-            <xsd:documentation>Gender type codes.</xsd:documentation>
-        </xsd:annotation>
-        <xsd:restriction base="xsd:string">
-            <xsd:enumeration value="MALE"/>
-            <xsd:enumeration value="FEMALE"/>
-            <xsd:enumeration value="CHILD"/>
-            <xsd:enumeration value="UNKNOWN"/>
-        </xsd:restriction>
-    </xsd:simpleType>
+<xsd:schema xmlns:tns="urn:CUSS-1.5/types/biometrics" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:oneIdTypes="http://www.materna.com/CUSS/OneIdTypes-1.0" targetNamespace="urn:CUSS-1.5/types/biometrics" elementFormDefault="qualified" attributeFormDefault="unqualified">
 
-    <xsd:complexType name="nameDataType">
-        <xsd:annotation>
-            <xsd:documentation>A raw or converted name. The raw name can be in elements name or fName and lName. The converted name is stored in the parts list, title, etc.</xsd:documentation>
-        </xsd:annotation>
-        <xsd:sequence>
-            <xsd:annotation>
-                <xsd:documentation>name data structure</xsd:documentation>
-            </xsd:annotation>
-            <xsd:element name="name" type="xsd:string" minOccurs="0">
-                <xsd:annotation>
-                    <xsd:documentation>full name</xsd:documentation>
-                </xsd:annotation>
-            </xsd:element>
-            <xsd:element name="fName" type="xsd:string" minOccurs="0">
-                <xsd:annotation>
-                    <xsd:documentation>first name.</xsd:documentation>
-                </xsd:annotation>
-            </xsd:element>
-            <xsd:element name="fNameParts" type="xsd:string" minOccurs="0" maxOccurs="unbounded">
-                <xsd:annotation>
-                    <xsd:documentation>first name in form of split up and normalized (“ä”-> “ae”, etc.) name parts.</xsd:documentation>
-                </xsd:annotation>
-            </xsd:element>
-            <xsd:element name="lName" type="xsd:string" minOccurs="0">
-                <xsd:annotation>
-                    <xsd:documentation>last name.</xsd:documentation>
-                </xsd:annotation>
-            </xsd:element>
-            <xsd:element name="lNameParts" type="xsd:string" minOccurs="0" maxOccurs="unbounded">
-                <xsd:annotation>
-                    <xsd:documentation>last name in form of split up and normalized (“ä”-> “ae”, etc.) name parts.</xsd:documentation>
-                </xsd:annotation>
-            </xsd:element>
-            <xsd:element name="gender" type="tns:genderType" minOccurs="0">
-                <xsd:annotation>
-                    <xsd:documentation>gender</xsd:documentation>
-                </xsd:annotation>
-            </xsd:element>
-        </xsd:sequence>
-    </xsd:complexType>
+  	<xsd:import namespace="http://www.materna.com/CUSS/OneIdTypes-1.0" schemaLocation="OneIdTypes.xsd"/>
 
-    <xsd:complexType name="passportData">
-        <xsd:annotation>
-            <xsd:documentation>Passport document data structure</xsd:documentation>
-        </xsd:annotation>
-        <xsd:sequence>
-            <xsd:element name="name" type="tns:nameDataType" minOccurs="0">
-                <xsd:annotation>
-                    <xsd:documentation>Name data structure</xsd:documentation>
-                </xsd:annotation>
-            </xsd:element>
-            <xsd:element name="documentNumber" type="xsd:string" minOccurs="0">
-                <xsd:annotation>
-                    <xsd:documentation>Document number</xsd:documentation>
-                </xsd:annotation>
-            </xsd:element>
-            <xsd:element name="expiryDate" type="xsd:date" minOccurs="0">
-                <xsd:annotation>
-                    <xsd:documentation>Expiry date of this document</xsd:documentation>
-                </xsd:annotation>
-            </xsd:element>
-            <xsd:element name="countryOfIssue" type="xsd:string" minOccurs="0">
-                <xsd:annotation>
-                    <xsd:documentation>ISO code of document issuing country</xsd:documentation>
-                </xsd:annotation>
-            </xsd:element>
-            <xsd:element name="cityOfIssue" type="xsd:string" minOccurs="0">
-                <xsd:annotation>
-                    <xsd:documentation>nameof issuing city</xsd:documentation>
-                </xsd:annotation>
-            </xsd:element>
-            <xsd:element name="dateOfBirth" type="xsd:date" minOccurs="0">
-                <xsd:annotation>
-                    <xsd:documentation>Date of birth</xsd:documentation>
-                </xsd:annotation>
-            </xsd:element>
-            <xsd:element name="nationality" type="xsd:string" minOccurs="0">
-                <xsd:annotation>
-                    <xsd:documentation>Document owner's nationality</xsd:documentation>
-                </xsd:annotation>
-            </xsd:element>
-            <xsd:element name="dateOfIssue" type="xsd:date" minOccurs="0">
-                <xsd:annotation>
-                    <xsd:documentation>Issue date of this document</xsd:documentation>
-                </xsd:annotation>
-            </xsd:element>
-            <xsd:element name="mrzTrack" type="xsd:string" minOccurs="0" maxOccurs="unbounded">
-                <xsd:annotation>
-                    <xsd:documentation>List of "machine readable zone" tracks</xsd:documentation>
-                </xsd:annotation>
-            </xsd:element>
-            <xsd:element name="passportImage" type="xsd:base64Binary" minOccurs="0">
-                <xsd:annotation>
-                    <xsd:documentation>Base64 encoded passport image data</xsd:documentation>
-                </xsd:annotation>
-            </xsd:element>
-        </xsd:sequence>
-    </xsd:complexType>
-
-    <xsd:complexType name="boardingPassData">
-        <xsd:annotation>
-            <xsd:documentation>Passport document data structure</xsd:documentation>
-        </xsd:annotation>
-        <xsd:sequence>
-            <xsd:element name="name" type="tns:nameDataType" minOccurs="0">
-                <xsd:annotation>
-                    <xsd:documentation>Name data structure</xsd:documentation>
-                </xsd:annotation>
-            </xsd:element>
-            <!-- flight related data -->
-            <xsd:element name="airlineCode" type="xsd:string" minOccurs="0">
-                <xsd:annotation>
-                    <xsd:documentation>2L-Airline code</xsd:documentation>
-                </xsd:annotation>
-            </xsd:element>
-            <xsd:element name="flightNumber" type="xsd:string" minOccurs="0">
-                <xsd:annotation>
-                    <xsd:documentation>Flight number in trimmed format, may include flight number suffix</xsd:documentation>
-                </xsd:annotation>
-            </xsd:element>
-            <xsd:element name="flightDate" type="xsd:date" minOccurs="0">
-                <xsd:annotation>
-                    <xsd:documentation>Full flight date</xsd:documentation>
-                </xsd:annotation>
-            </xsd:element>
-            <xsd:element name="origin" type="xsd:string" minOccurs="0">
-                <xsd:annotation>
-                    <xsd:documentation>Origin 3L-code</xsd:documentation>
-                </xsd:annotation>
-            </xsd:element>
-            <xsd:element name="destination" type="xsd:string" minOccurs="0">
-                <xsd:annotation>
-                    <xsd:documentation>Destination 3L-code</xsd:documentation>
-                </xsd:annotation>
-            </xsd:element>
-            <!-- pax related data -->
-            <xsd:element name="sequenceNumber" type="xsd:string" minOccurs="0">
-                <xsd:annotation>
-                    <xsd:documentation>Boarding number or standby number for this segment, if any. The type is determined by the checkinStatus.</xsd:documentation>
-                </xsd:annotation>
-            </xsd:element>
-            <xsd:element name="pnr" type="xsd:string" minOccurs="0">
-                <xsd:annotation>
-                    <xsd:documentation>PNR code (file key) for this segment.</xsd:documentation>
-                </xsd:annotation>
-            </xsd:element>
-            <xsd:element name="compartmentCode" type="xsd:string" minOccurs="0">
-                <xsd:annotation>
-                    <xsd:documentation>Passenger's compartment code (e.g. 'F')</xsd:documentation>
-                </xsd:annotation>
-            </xsd:element>
-            <xsd:element name="seat" type="xsd:string" minOccurs="0">
-                <xsd:annotation>
-                    <xsd:documentation>Passenger's seat (e.g. '015A)</xsd:documentation>
-                </xsd:annotation>
-            </xsd:element>
-            <xsd:element name="barcodeString" type="xsd:string" minOccurs="0">
-                <xsd:annotation>
-                    <xsd:documentation>List of "machine readable zone" tracks</xsd:documentation>
-                </xsd:annotation>
-            </xsd:element>
-        </xsd:sequence>
-    </xsd:complexType>
-    
-    <xsd:complexType name="pax">
-        <xsd:annotation>
-            <xsd:documentation></xsd:documentation>
-        </xsd:annotation>
-        <xsd:sequence>
-            <xsd:element name="id" type="xsd:string" minOccurs="0">
-                <xsd:annotation>
-                    <xsd:documentation>Passenger id, if applicable</xsd:documentation>
-                </xsd:annotation>
-            </xsd:element>
-            <xsd:element name="hostNativeId" type="xsd:string" minOccurs="0">
-                <xsd:annotation>
-                    <xsd:documentation>Host specific passenger id, if applicable</xsd:documentation>
-                </xsd:annotation>
-            </xsd:element>
-            <xsd:element name="name" type="tns:nameDataType" minOccurs="0">
-                <xsd:annotation>
-                    <xsd:documentation>Name data structure</xsd:documentation>
-                </xsd:annotation>
-            </xsd:element>
-            <xsd:element name="passport" type="tns:passportData" minOccurs="0">
-                <xsd:annotation>
-                    <xsd:documentation>TBC</xsd:documentation>
-                </xsd:annotation>
-            </xsd:element>
-            <xsd:element name="boardingPass" type="tns:boardingPassData" minOccurs="0">
-                <xsd:annotation>
-                    <xsd:documentation>TBC</xsd:documentation>
-                </xsd:annotation>
-            </xsd:element>
-            <xsd:element name="facialImage" type="xsd:base64Binary" minOccurs="0">
-                <xsd:annotation>
-                    <xsd:documentation>Base64 encoded facial image data. Required in case of identification via facial image</xsd:documentation>
-                </xsd:annotation>
-            </xsd:element>
-        </xsd:sequence>
-    </xsd:complexType>
-    
     <xsd:simpleType name="score">
         <xsd:annotation>
             <xsd:documentation>Comparision or match score</xsd:documentation>
@@ -283,9 +71,9 @@
                     <xsd:documentation>Error occured (only applicable for response commands)</xsd:documentation>
                 </xsd:annotation>
             </xsd:enumeration>
-            <xsd:enumeration value="identify">
+            <xsd:enumeration value="verify">
                 <xsd:annotation>
-                    <xsd:documentation>Identifies a passenger. What passenger details are retured depends on the implementation</xsd:documentation>
+                    <xsd:documentation>Verifies if a passenger is already enrolled by given boarding pass data or if the the given camera image and boarding pass data is matching the enrolled passenger (supported by NEC service handler only)</xsd:documentation>
                 </xsd:annotation>
             </xsd:enumeration>
         </xsd:restriction>
@@ -450,7 +238,7 @@
             <xsd:documentation>Data element for command enrollPerson</xsd:documentation>
         </xsd:annotation>
         <xsd:sequence>
-            <xsd:element name="identCode" type="xsd:base64Binary">
+            <xsd:element name="identCode" type="xsd:base64Binary" minOccurs="0">
                 <xsd:annotation>
                     <xsd:documentation>Base64 encoded identification data (e.g. barcode)</xsd:documentation>
                 </xsd:annotation>
@@ -597,7 +385,17 @@
             </xsd:enumeration>
             <xsd:enumeration value="notIdentified">
                 <xsd:annotation>
-                    <xsd:documentation>Passenger is unknown and cannt be identifies (i.e. not enrolled before)</xsd:documentation>
+                    <xsd:documentation>Passenger is unknown and cannot be identified (i.e. not enrolled before)</xsd:documentation>
+                </xsd:annotation>
+            </xsd:enumeration>
+            <xsd:enumeration value="poorCameraImage">
+                <xsd:annotation>
+                    <xsd:documentation>Provided camera image is insufficient for the required function</xsd:documentation>
+                </xsd:annotation>
+            </xsd:enumeration>
+            <xsd:enumeration value="poorDocumentImage">
+                <xsd:annotation>
+                    <xsd:documentation>Provided document image (e.g scanned from passport) is insufficient for the required function</xsd:documentation>
                 </xsd:annotation>
             </xsd:enumeration>
         </xsd:restriction>
@@ -725,6 +523,7 @@
                     <xsd:documentation>Unique identifier for an identified person. Only applicable for a command response to getIdentifiedBiometricImage</xsd:documentation>
                 </xsd:annotation>
             </xsd:element>
+            <!-- Deprecated -->
             <xsd:element name="identCode" type="xsd:base64Binary" minOccurs="0">
                 <xsd:annotation>
                     <xsd:documentation>Identifier which was enrolled for this person as (*potentially crypted*) base64 representation (e.g. barcode, passport, xml...)</xsd:documentation>
@@ -802,7 +601,7 @@
             <xsd:documentation>biometric root type</xsd:documentation>
         </xsd:annotation>
         <xsd:sequence>
-            <xsd:element name="pax" type="tns:pax" minOccurs="0">
+            <xsd:element name="pax" type="oneIdTypes:passenger" minOccurs="0">
                 <xsd:annotation>
                     <xsd:documentation>passenger data, if applicable</xsd:documentation>
                 </xsd:annotation>
diff --git a/SelfServiceCommon/Massai/idl/Cuss/OneIdTypes.xsd b/SelfServiceCommon/Massai/idl/Cuss/OneIdTypes.xsd
new file mode 100644
index 00000000..e97b194c
--- /dev/null
+++ b/SelfServiceCommon/Massai/idl/Cuss/OneIdTypes.xsd
@@ -0,0 +1,268 @@
+<?xml version="1.0" encoding="utf-8"?>
+<xsd:schema xmlns:oneIdTypes="http://www.materna.com/CUSS/OneIdTypes-1.0" xmlns:xsd="http://www.w3.org/2001/XMLSchema" targetNamespace="http://www.materna.com/CUSS/OneIdTypes-1.0">
+    <xsd:simpleType name="genderType">
+        <xsd:annotation>
+            <xsd:documentation>Gender type codes.</xsd:documentation>
+        </xsd:annotation>
+        <xsd:restriction base="xsd:string">
+            <xsd:enumeration value="MALE"/>
+            <xsd:enumeration value="FEMALE"/>
+            <xsd:enumeration value="CHILD"/>
+            <xsd:enumeration value="UNKNOWN"/>
+        </xsd:restriction>
+    </xsd:simpleType>
+
+    <xsd:complexType name="nameData">
+        <xsd:annotation>
+            <xsd:documentation>A raw or converted name. The raw name can be in elements name or fName and lName. The converted name is stored in the parts list, title, etc.</xsd:documentation>
+        </xsd:annotation>
+        <xsd:sequence>
+            <xsd:annotation>
+                <xsd:documentation>name data structure</xsd:documentation>
+            </xsd:annotation>
+            <xsd:element name="name" type="xsd:string" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>full name</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="fName" type="xsd:string" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>first name.</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="fNameParts" type="xsd:string" minOccurs="0" maxOccurs="unbounded">
+                <xsd:annotation>
+                    <xsd:documentation>first name in form of split up and normalized (“ä”-> “ae”, etc.) name parts.</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="lName" type="xsd:string" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>last name.</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="lNameParts" type="xsd:string" minOccurs="0" maxOccurs="unbounded">
+                <xsd:annotation>
+                    <xsd:documentation>last name in form of split up and normalized (“ä”-> “ae”, etc.) name parts.</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="gender" type="oneIdTypes:genderType" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>gender</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+        </xsd:sequence>
+    </xsd:complexType>
+
+    <xsd:complexType name="idDocumentData">
+        <xsd:annotation>
+            <xsd:documentation>ID / passport document data structure</xsd:documentation>
+        </xsd:annotation>
+        <xsd:sequence>
+            <xsd:element name="name" type="oneIdTypes:nameData" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>Name data structure</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="documentNumber" type="xsd:string" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>Document number</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="expiryDate" type="xsd:date" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>Expiry date of this document</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="countryOfIssue" type="xsd:string" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>ISO code of document issuing country</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="cityOfIssue" type="xsd:string" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>nameof issuing city</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="dateOfBirth" type="xsd:date" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>Date of birth</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="nationality" type="xsd:string" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>Document owner's nationality</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="dateOfIssue" type="xsd:date" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>Issue date of this document</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="mrzData" minOccurs="0" maxOccurs="3">
+                <xsd:annotation>
+                    <xsd:documentation>List of "machine readable zone" tracks</xsd:documentation>
+                </xsd:annotation>
+                <xsd:complexType>
+                    <xsd:simpleContent>
+                        <xsd:extension base="xsd:string">
+                            <xsd:attribute name="mrz-track" use="required">
+                                <xsd:simpleType>
+                                    <xsd:restriction base="xsd:unsignedInt">
+                                        <xsd:enumeration value="1"/>
+                                        <xsd:enumeration value="2"/>
+                                        <xsd:enumeration value="3"/>
+                                    </xsd:restriction>
+                                </xsd:simpleType>
+                            </xsd:attribute>
+                        </xsd:extension>
+                    </xsd:simpleContent>
+                </xsd:complexType>
+            </xsd:element>
+            <xsd:element name="documentImage" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>Base64 encoded document image data</xsd:documentation>
+                </xsd:annotation>
+                <xsd:complexType>
+                    <xsd:simpleContent>
+                        <xsd:extension base="xsd:base64Binary">
+                            <xsd:attribute name="data-format" use="optional">
+                                <xsd:annotation>
+                                    <xsd:documentation>Image data format.</xsd:documentation>
+                                </xsd:annotation>
+                                <xsd:simpleType>
+                                    <xsd:restriction base="xsd:string">
+                                        <xsd:enumeration value="bmp"/>
+                                        <xsd:enumeration value="jpg"/>
+                                        <xsd:enumeration value="png"/>
+                                    </xsd:restriction>
+                                </xsd:simpleType>
+                            </xsd:attribute>
+                        </xsd:extension>
+                    </xsd:simpleContent>
+                </xsd:complexType>
+            </xsd:element>
+        </xsd:sequence>
+    </xsd:complexType>
+
+    <xsd:complexType name="boardingPassData">
+        <xsd:annotation>
+            <xsd:documentation>Passport document data structure</xsd:documentation>
+        </xsd:annotation>
+        <xsd:sequence>
+            <xsd:element name="name" type="oneIdTypes:nameData" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>Name data structure</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <!-- flight related data -->
+            <xsd:element name="airlineCode" type="xsd:string" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>2L-Airline code</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="flightNumber" type="xsd:string" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>Flight number in trimmed format, may include flight number suffix</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="flightDate" type="xsd:date" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>Full flight date</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="origin" type="xsd:string" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>Origin 3L-code</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="destination" type="xsd:string" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>Destination 3L-code</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <!-- pax related data -->
+            <xsd:element name="sequenceNumber" type="xsd:string" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>Boarding number or standby number for this segment, if any. The type is determined by the checkinStatus.</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="pnr" type="xsd:string" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>PNR code (file key) for this segment.</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="compartmentCode" type="xsd:string" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>Passenger's compartment code (e.g. 'F')</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="seat" type="xsd:string" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>Passenger's seat (e.g. '015A)</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="barcodeString" type="xsd:string" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>List of "machine readable zone" tracks</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+        </xsd:sequence>
+    </xsd:complexType>
+
+    <xsd:complexType name="passenger">
+        <xsd:annotation>
+            <xsd:documentation></xsd:documentation>
+        </xsd:annotation>
+        <xsd:sequence>
+            <xsd:element name="id" type="xsd:string" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>Passenger id, if applicable</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="hostNativeId" type="xsd:string" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>Host specific passenger id, if applicable</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="name" type="oneIdTypes:nameData" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>Name data structure</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="idDocument" type="oneIdTypes:idDocumentData" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>TBC</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="boardingPass" type="oneIdTypes:boardingPassData" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>TBC</xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="facialImage" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>Base64 encoded facial image data. Required in case of identification via facial image</xsd:documentation>
+                </xsd:annotation>
+                <xsd:complexType>
+                    <xsd:simpleContent>
+                        <xsd:extension base="xsd:base64Binary">
+                            <xsd:attribute name="data-format" use="optional">
+                                <xsd:annotation>
+                                    <xsd:documentation>Image data format.</xsd:documentation>
+                                </xsd:annotation>
+                                <xsd:simpleType>
+                                    <xsd:restriction base="xsd:string">
+                                        <xsd:enumeration value="bmp"/>
+                                        <xsd:enumeration value="jpg"/>
+                                        <xsd:enumeration value="png"/>
+                                    </xsd:restriction>
+                                </xsd:simpleType>
+                            </xsd:attribute>
+                        </xsd:extension>
+                    </xsd:simpleContent>
+                </xsd:complexType>
+            </xsd:element>
+        </xsd:sequence>
+    </xsd:complexType>
+
+</xsd:schema>
-- 
2.41.0.windows.1

